ciao-4.11/contrib/bin/download_obsid_caldb

```diff
@@ -476,64 +472,66 @@ def get_all_products():
 
     This is a different way to find the caldb products -- it looks for
     all the index files and gets the unique set of CAL_CNAM values.
-    
-    The problem with this is that the same single fits file can have 
+
+    The problem with this is that the same single fits file can have
     multiple extensions, and each extension may have a different
     CAL_CNAM (eg ACIS CTI file p2_resp files).  This then makes the retreival
     look more messy since the file will be 'skipped' even in a clean caldb tree.
     Guess we could filter out certain ones -- but not sure any more/less
     complicated then the get_by_tool method below
     """
     from pycrates import read_file
-        
+
     cidx = os.environ["CALDBCONFIG"]
     cdir = os.environ["CALDB"]
-    with open( cidx, "r+") as fp:
+    with open(cidx, "r+") as fp:
         lines = fp.readlines()
-    
+
     chandra = [x for x in lines if x.startswith("CHANDRA")]
-    ciao = [x for x in chandra if x.split()[1] not in  ['EPHIN', 'PCAD', 'SIM' ]]
+    ciao = [x for x in chandra if x.split()[1] not in ['EPHIN', 'PCAD', 'SIM']]
     ciao = [x for x in ciao if 'pimms' not in x.lower()]
-    ciao = [ x.replace(" CALDB ", " {} ".format(cdir)) for x in ciao]
-    idxfiles = [ os.path.join( x.split()[2], os.path.join( x.split()[3],x.split()[4] )) for x in ciao ]
+    ciao = [x.replace(" CALDB ", " {} ".format(cdir)) for x in ciao]
+    idxfiles = [os.path.join(x.split()[2],
```

Comment:
Not related to this PR.

This can be simplified ; `os.path.join` takes multiple dirs.

Reply:
I re-wrote this to be a single loop, rather than a set of list comprehensions. I think it's easier to see the logic of what is/isn't included. Assuming I haven't botched something up (as the logic checks are now inverted).